site/tests/conftest.py

```diff
@@ -84,3 +109,303 @@ def client_with_login(client, users):
+def resource_type_v(app):
+    """Resource type vocabulary record."""
+    vocabulary_service.create_type(system_identity, "resourcetypes", "rsrct")
+    vocab = vocabulary_service.create(
+        system_identity,
+        {
+            "id": "image-photo",
+            "tags": ["depositable", "linkable"],
+            "type": "resourcetypes",
+        },
+    )
+    Vocabulary.index.refresh()
+    return vocab
+
+
+#
+# User fixture helper - move to pytest-invenio
+#
+class UserFixture_:
+    """A user fixture for easy test user creation."""
+
+    def __init__(self, email=None, password=None):
+        """Constructor."""
+        self._email = email
+        self._password = password
+        self._identity = None
+        self._user = None
+
+    #
+    # Creation
+    #
+    def create(self, app, db):
+        """Create the user."""
+        with db.session.begin_nested():
+            datastore = app.extensions["security"].datastore
+            user = datastore.create_user(
+                email=self.email,
+                password=hash_password(self.password),
+            )
+        db.session.commit()
+        self._user = user
+        return self
+
+    #
+    # Properties
+    #
+    @property
+    def user(self):
+        """Get the user."""
+        return self._user
+
+    @property
+    def email(self):
+        """Get the email."""
+        return self._email
+
+    @property
+    def password(self):
+        """Get the password."""
+        return self._password
+
+    #
+    # App context helpers
+    #
+    @property
+    def identity(self):
+        """Create identity for the user."""
+        if self._identity is None:
+            # Simulate a full login
+            assert login_user(self.user)
+            self._identity = deepcopy(g.identity)
+            # Clean up - we just want the identity object.
+            logout_user()
+        return self._identity
+
+    #
+    # Test client helpers
+    #
+    def login(self, client, logout_first=False):
+        """Login the given client."""
+        return self._login(client, "/", logout_first)
+
+    def logout(self, client):
+        """Logout the given client."""
+        return self._logout(client, "/")
+
+    def _login(self, client, base_path, logout):
+        """Login the given client."""
+        if logout:
+            self._logout(client, base_path)
+        res = client.post(
+            f"{base_path}login",
+            data=dict(email=self.email, password=self.password),
+            environ_base={"REMOTE_ADDR": "127.0.0.1"},
+            follow_redirects=True,
+        )
+        assert res.status_code == 200
+        return client
+
+    def _logout(self, client, base_path):
+        """Logout the client."""
+        res = client.get(f"{base_path}logout")
+        assert res.status_code < 400
+        return client
+
+
+@pytest.fixture(scope="session")
+def UserFixture():
+    """Class to create user fixtures from."""
+    return UserFixture_
```

Comment:
This was already moved to pytest-invenio, see [here](https://github.com/inveniosoftware/pytest-invenio/blob/master/pytest_invenio/fixtures.py#L855). I think it can be removed.